File: java/TestCode.java
import java.util.Arrays;
import java.util.List;

public class TestCode {
    public static int solution(int[] numbers) {
        System.out.println(Arrays.toString(numbers));
        return backtrack(0, 0, 0, numbers);
    }

    static int backtrack(int total, int sum, int start, int[] numbers) {
        if (start >= numbers.length) {
            if (sum % 2 == 0) {
                return total + 1;
            }
            return total;
        }
        int n = numbers[start];
        while (n != 0) {
            System.out.println(n);
            int cur = n % 10;
            total = backtrack(total, start + cur, start + 1, numbers);
            n = n / 10;
        }
        return total;
    }

    public static void main(String[] args) {
        // You can add more test cases here
        System.out.println(solution(new int[] { 123, 456, 789 }) == 14);
        // System.out.println(solution(new int[] { 123456789 }) == 4);
        // System.out.println(solution(new int[] { 14329, 7568 }) == 10);
    }
}

File: java/TestCode.js
function solution(numbers) {
    console.log(numbers.toString());
    return backtrack(0, 0, 0, numbers);
}

function backtrack(total, sum, start, numbers) {
    if (start >= numbers.length) {
        if (sum % 2 == 0) {
            console.log(sum);
            return total + 1;
        }
        return total;
    }
    let n = numbers[start];
    while (n != 0) {
        let cur = n % 10;
        // console.log(cur);
        total = backtrack(total, sum + cur, start + 1, numbers);
        n = Math.floor(n / 10);
    }
    return total;
}

// 测试用例
console.log(solution([123, 456, 789]) === 14);
// console.log(solution([123456789]) === 4);
// console.log(solution([14329, 7568]) === 10);
